From 887c1367ba633357abe13dde39363f2a9da4c6c6 Mon Sep 17 00:00:00 2001
From: Holger Behrens <holger.behrens@windriver.com>
Date: Mon, 17 Nov 2014 16:39:21 +0100
Subject: [[PATCH] configure.ac: fix typo libperscommon

Signed-off-by: Holger Behrens <holger.behrens@windriver.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 5207334..6e1b2dd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -113,7 +113,7 @@ PKG_CHECK_MODULES([DBUS], [dbus-1 >= 1.4.10])
 PKG_CHECK_MODULES([DLT], [automotive-dlt >= 2.2.0])	
 PKG_CHECK_MODULES([ZLIB], [zlib >= 1.2.5])		
 PKG_CHECK_MODULES([ARCHIVELIB], [libarchive >= 3.0.4])	
-PKG_CHECK_MODULES([PERSCOMMON], [libperscommon >= 1.0.1])
+PKG_CHECK_MODULES([PERSCOMMON], [persistence_client_library >= 1.0.1])
 PKG_CHECK_MODULES([JSON], [json >= 0.9])
 PKG_CHECK_MODULES([NSM], [node-state-manager])
 PKG_CHECK_MODULES([SYSTEMD], [libsystemd-daemon >= 37 ])
-- 
2.1.3

