diff --git a/configure.ac b/configure.ac
index f1fde52..44a1045 100644
--- a/configure.ac
+++ b/configure.ac
@@ -116,7 +116,7 @@ PKG_CHECK_MODULES([ARCHIVELIB], [libarchive >= 3.0.4])
 PKG_CHECK_MODULES([PERSCOMMON], [libperscommon >= 1.0.1])
 PKG_CHECK_MODULES([JSON], [json >= 0.9])
 PKG_CHECK_MODULES([NSM], [node-state-manager])
-PKG_CHECK_MODULES([SYSTEMD], [libsystemd-daemon >= 37 ])
+PKG_CHECK_MODULES([SYSTEMD], [libsystemd >= 37 ])
 
 AC_SUBST(DEPS_CFLAGS)
 AC_SUBST(DEPS_LIBS)
