Index: libitzam-6.0.4/configure.ac
===================================================================
--- libitzam-6.0.4.orig/configure.ac	2011-09-17 20:16:04.000000000 +0300
+++ libitzam-6.0.4/configure.ac	2014-02-24 13:43:02.000000000 +0200
@@ -1,7 +1,7 @@
 AC_INIT(src/itzam.h)
 AC_CONFIG_MACRO_DIR([m4])
 
-GENERIC_LIBRARY_NAME=libitzam
+GENERIC_LIBRARY_NAME=itzam
 
 #release versioning
 GENERIC_MAJOR_VERSION=6
