From 0f5eca77f55f7236b51016e0d7573c078c895640 Mon Sep 17 00:00:00 2001
From: Holger Behrens <holger.behrens@windriver.com>
Date: Mon, 17 Nov 2014 18:29:45 +0100
Subject: [[PATCH] persComTypes.h: Bug 251 work around

Signed-off-by: Holger Behrens <holger.behrens@windriver.com>
---
 inc/protected/persComTypes.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/inc/protected/persComTypes.h b/inc/protected/persComTypes.h
index 47a6664..7a312b4 100644
--- a/inc/protected/persComTypes.h
+++ b/inc/protected/persComTypes.h
@@ -83,7 +83,7 @@ typedef __s64 INT64;
 typedef __u8  uint8_t;
 typedef __u16 uint16_t;
 typedef __u32 uint32_t;
-typedef __u64 uint64_t;
+//typedef __u64 uint64_t;
 #endif
 
 /**
@@ -203,7 +203,7 @@ typedef sint32_t* psint32_t;
 /**
  * \brief Pointer to UNSIGNED-64-Bit
  */
-typedef uint64_t* puint64_t;
+//typedef uint64_t* puint64_t;
 /**
  * \brief Pointer to SIGNED-64-Bit
  */
-- 
2.1.3

