From d4aa6cf7344c3d1b244fdddfe95d8300f35aa157 Mon Sep 17 00:00:00 2001
From: Holger Behrens <holger.behrens@windriver.com>
Date: Mon, 17 Nov 2014 17:08:08 +0100
Subject: [[PATCH] configure.ac: fix typo

Signed-off-by: Holger Behrens <holger.behrens@windriver.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index badd5c6..88e8a63 100644
--- a/configure.ac
+++ b/configure.ac
@@ -25,7 +25,7 @@ m4_define([PERSCOMMON_LIB_VERSION_N],[1001001])
 dnl ***************************
 dnl *** Initialize autoconf ***
 dnl ***************************
-AC_INIT([persistence-common],[LIBPERSOCOMMON_VERSION_S()])
+AC_INIT([persistence-common],[PERSCOMMON_PACKAGE_VERSION_S])
 AC_COPYRIGHT([Copyright (c) 2012 Continental Automotive GmbH])
 AC_PREREQ([2.50])
 
-- 
2.1.3

