Index: git/systemd/CMakeLists.txt
===================================================================
--- git.orig/systemd/CMakeLists.txt	2013-12-03 21:32:52.000000000 +0200
+++ git/systemd/CMakeLists.txt	2013-12-03 21:33:26.000000000 +0200
@@ -16,7 +16,7 @@
 
 if(WITH_SYSTEMD)
 
-    set(SYSTEMD_CONFIGURATIONS_FILES_DIR ${CMAKE_INSTALL_PREFIX}/lib/systemd/system )
+    set(SYSTEMD_CONFIGURATIONS_FILES_DIR /lib/systemd/system )
         
     if(WITH_SYSTEMD_WATCHDOG)
         set( DLT_WatchdogSec 2 )
