--- dbus-1.10.10/dbus/dbus-message.c.orig	2017-01-26 20:32:02.456180239 +0900
+++ dbus-1.10.10/dbus/dbus-message.c	2017-01-26 20:32:27.288180875 +0900
@@ -4128,7 +4128,7 @@
      beginning. This sucks a bit, however unless SCM_RIGHTS is fixed
      there is no better way. */
 
-  if (loader->n_unix_fds_allocated < loader->max_message_unix_fds)
+  if ((int)(loader->n_unix_fds_allocated) < loader->max_message_unix_fds)
     {
       int *a = dbus_realloc(loader->unix_fds,
                             loader->max_message_unix_fds * sizeof(loader->unix_fds[0]));
