--- dbus-1.10.6/dbus/dbus-marshal-recursive.c.orig	2016-06-30 15:25:51.655302658 +0900
+++ dbus-1.10.6/dbus/dbus-marshal-recursive.c	2016-06-30 15:26:51.559300885 +0900
@@ -989,7 +989,7 @@
 _dbus_type_reader_recurse (DBusTypeReader *reader,
                            DBusTypeReader *sub)
 {
-  const DBusTypeReaderClass *klass;
+  const DBusTypeReaderClass *klass = NULL;
   int t;
 
   t = _dbus_first_type_in_signature (reader->type_str, reader->type_pos);
--- dbus-1.10.6/dbus/dbus-message.c.orig	2016-07-01 20:24:23.884214520 +0900
+++ dbus-1.10.6/dbus/dbus-message.c	2016-07-01 20:24:49.708213755 +0900
@@ -4091,7 +4091,7 @@
      beginning. This sucks a bit, however unless SCM_RIGHTS is fixed
      there is no better way. */
 
-  if (loader->n_unix_fds_allocated < loader->max_message_unix_fds)
+  if (loader->n_unix_fds_allocated < (unsigned long)(loader->max_message_unix_fds))
     {
       int *a = dbus_realloc(loader->unix_fds,
                             loader->max_message_unix_fds * sizeof(loader->unix_fds[0]));
