diff -ur bluez-5.48.bak/src/device.c bluez-5.48/src/device.c
--- bluez-5.48.bak/src/device.c	2018-08-20 14:34:33.895522420 +0200
+++ bluez-5.48/src/device.c	2018-08-20 14:35:13.699483187 +0200
@@ -5219,6 +5219,10 @@
 	return device->path;
 }
 
+const char *btd_device_get_path(const struct btd_device *device) {
+	return device_get_path(device);
+}
+
 gboolean device_is_temporary(struct btd_device *device)
 {
 	return device->temporary;
diff -ur bluez-5.48.bak/src/device.h bluez-5.48/src/device.h
--- bluez-5.48.bak/src/device.h	2018-08-20 14:34:33.895522420 +0200
+++ bluez-5.48/src/device.h	2018-08-20 14:35:13.699483187 +0200
@@ -88,6 +88,7 @@
 struct btd_adapter *device_get_adapter(struct btd_device *device);
 const bdaddr_t *device_get_address(struct btd_device *device);
 const char *device_get_path(const struct btd_device *device);
+const char *btd_device_get_path(const struct btd_device *device);
 gboolean device_is_temporary(struct btd_device *device);
 bool device_is_connectable(struct btd_device *device);
 bool device_is_paired(struct btd_device *device, uint8_t bdaddr_type);
