Index: Mesa-8.0.1/src/glsl/builtins/profiles/OES_EGL_image_external.frag
===================================================================
--- Mesa-8.0.1.orig/src/glsl/builtins/profiles/OES_EGL_image_external.frag	2012-03-20 18:22:02.645070696 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,6 +0,0 @@
-#version 100
-#extension GL_OES_EGL_image_external : enable
-
-vec4 texture2D(samplerExternalOES sampler, vec2 coord);
-vec4 texture2DProj(samplerExternalOES sampler, vec3 coord);
-vec4 texture2DProj(samplerExternalOES sampler, vec4 coord);
Index: Mesa-8.0.1/src/glsl/builtins/profiles/OES_EGL_image_external.vert
===================================================================
--- Mesa-8.0.1.orig/src/glsl/builtins/profiles/OES_EGL_image_external.vert	2012-03-20 18:22:02.661070696 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,6 +0,0 @@
-#version 100
-#extension GL_OES_EGL_image_external : enable
-
-vec4 texture2D(samplerExternalOES sampler, vec2 coord);
-vec4 texture2DProj(samplerExternalOES sampler, vec3 coord);
-vec4 texture2DProj(samplerExternalOES sampler, vec4 coord);
