Index: git/LayerManagerExamples/LayerManagerControl/src/commands.cpp
===================================================================
--- git.orig/LayerManagerExamples/LayerManagerControl/src/commands.cpp	2012-07-12 16:51:49.695274924 +0300
+++ git/LayerManagerExamples/LayerManagerControl/src/commands.cpp	2012-07-12 16:52:03.231275452 +0300
@@ -21,6 +21,7 @@
 #include "ExpressionInterpreter.h"
 #include <iostream>
 #include <signal.h> // signal
+#include <unistd.h>
 
 using namespace std;
 
