Subject: [PATCH] Add the processing which has notice of click event from compositor

Signed-off-by: Nobuhiko Tanibata <NOBUHIKO_TANIBATA at xddp.denso.co.jp>
---
 ivi-shell/ivi-shell.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/ivi-shell/ivi-shell.c b/ivi-shell/ivi-shell.c
index 2f2e4fa..933e878 100644
--- a/ivi-shell/ivi-shell.c
+++ b/ivi-shell/ivi-shell.c
@@ -36,6 +36,7 @@
 #include <dlfcn.h>
 #include <limits.h>
 #include <assert.h>
+#include <linux/input.h>
 
 #include "ivi-shell.h"
 #include "ivi-application-server-protocol.h"
@@ -385,6 +386,21 @@
 	return result;
 }
 
+static void
+click_to_activate_binding(struct weston_seat *seat,
+                          uint32_t time, uint32_t button,
+                          void *data)
+{
+	if (seat->pointer->grab != &seat->pointer->default_grab) {
+		return;
+	}
+	if (seat->pointer->focus == NULL) {
+		return;
+	}
+
+	weston_surface_activate(seat->pointer->focus->surface, seat);
+}
+
 /*
  * Initialization of ivi-shell.
  */
@@ -424,6 +440,8 @@
 		return -1;
 	}
 
+	weston_compositor_add_button_binding(compositor, BTN_LEFT, 0, click_to_activate_binding, shell);
+
 	free(setting.ivi_module);
 	return 0;
 }
1.7.11.7

BR,
Nobuhiko Tanibata
