diff -urN Mesa-10.4.4.orig/src/egl/main/egldriver.c Mesa-10.4.4/src/egl/main/egldriver.c
--- Mesa-10.4.4.orig/src/egl/main/egldriver.c	2015-02-07 09:09:51.000000000 +0900
+++ Mesa-10.4.4/src/egl/main/egldriver.c	2015-09-26 11:36:43.938445294 +0900
@@ -518,6 +518,19 @@
 
 
 /**
+ * Add egl_gallium to the module array.
+ */
+static void
+_eglAddGalliumDriver(void)
+{
+#ifndef _EGL_BUILT_IN_DRIVER_GALLIUM
+   void *external = (void *) "egl_gallium";
+   _eglPreloadForEach(_eglGetSearchPath(), _eglLoaderFile, external);
+#endif
+}
+
+
+/**
  * Add built-in drivers to the module array.
  */
 static void
@@ -549,6 +562,7 @@
        * Add other drivers only when EGL_DRIVER is not set.  The order here
        * decides the priorities.
        */
+      _eglAddGalliumDriver();
       _eglAddBuiltInDrivers();
    }
 
