Index: git/include/configs/omap3_beagle.h
===================================================================
--- git.orig/include/configs/omap3_beagle.h	2012-01-20 02:19:54.582012450 +0200
+++ git/include/configs/omap3_beagle.h	2012-01-20 02:20:33.534011227 +0200
@@ -195,7 +195,7 @@
 							/* partition */
 
 /* Environment information */
-#define CONFIG_BOOTDELAY		10
+#define CONFIG_BOOTDELAY		0
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"loadaddr=0x82000000\0" \
