Index: eglibc-2_16/libc/sysdeps/gnu/configure
===================================================================
--- libc/sysdeps/gnu/configure.orig	2012-08-13 13:24:54.643463697 +0300
+++ libc/sysdeps/gnu/configure	2012-08-13 13:25:42.231463057 +0300
@@ -37,6 +37,7 @@
   else
     libc_cv_localstatedir=$localstatedir
    fi
-  libc_cv_rootsbindir=/sbin
+  test -n "$libc_cv_rootsbindir" || \
+  libc_cv_rootsbindir="/sbin"
   ;;
 esac
