Index: git/weston-ivi-shell/src/ivi-layout-export.h
===================================================================
--- git.orig/weston-ivi-shell/src/ivi-layout-export.h	2014-07-10 13:49:00.274877329 +0300
+++ git/weston-ivi-shell/src/ivi-layout-export.h	2014-07-10 13:50:52.690874042 +0300
@@ -49,7 +49,7 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#include "compositor.h"
+#include <weston/compositor.h>
 #include "ivi-layout.h"
 
 struct ivi_layout_SurfaceProperties
Index: git/weston-ivi-shell/src/ivi-layout.h
===================================================================
--- git.orig/weston-ivi-shell/src/ivi-layout.h	2014-07-10 13:49:00.274877329 +0300
+++ git/weston-ivi-shell/src/ivi-layout.h	2014-07-10 13:51:02.802873746 +0300
@@ -49,7 +49,7 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#include "compositor.h"
+#include <weston/compositor.h>
 
 struct ivi_layout_surface;
 
