Index: git/weston-ivi-shell/CMakeLists.txt
===================================================================
--- git.orig/weston-ivi-shell/CMakeLists.txt
+++ git/weston-ivi-shell/CMakeLists.txt
@@ -49,7 +49,7 @@ link_directories(
     ${CAIRO_LIBRARY_DIRS}
     ${WESTON_LIBRARY_DIRS}
     ${PIXMAN_LIBRARY_DIRS}
-    ${WESTON_LIBDIR}/weston
+    =${WESTON_LIBDIR}/weston
     ${EGL_LIBRARY_DIRS}
     ${GLES_LIBRARY_DIRS}
 )
