From 7ae0a0dc56381b3b9410f16247f3fcaa1b25989b Mon Sep 17 00:00:00 2001
From: Nobuhiko Tanibata <NOBUHIKO_TANIBATA@denso.co.jp>
Date: Wed, 29 Jan 2014 15:53:59 +0900
Subject: [PATCH 01/31] README modifed for ivi-shell

---
 README | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/README b/README
index 3821e5d..cf7eca5 100644
--- a/README
+++ b/README
@@ -1,3 +1,20 @@
+ivi-shell addon:
+
+ivi-shell proposes two features
+- Common layout library for surface, which allow ivi-shell developer
+  to develop own shell, linking Common layout library.
+  For the time being, the library refers Genivi ilm interface.
+
+  http://projects.genivi.org/ivi-layer-management/
+
+- Extension protocal; ivi-application to tie wl_surface and a given ID.
+  With this ID, shell can identify which wl_surface is drawn by which
+  application. In in-vehicle infortainment system, a shell has to update
+  a property of a wl_surface. E.g. there may be a use case when vehicle
+  starts to move, the wl_surface drawn by Car navigation is expected to
+  move top of surfaces.
+
+-----------------------------
 Weston
 
 Weston is the reference implementation of a Wayland compositor, and a
-- 
1.9.3

