From 9f5ea45fed46102e9045311936481cdb2c13fced Mon Sep 17 00:00:00 2001
From: Nobuhiko Tanibata <NOBUHIKO_TANIBATA@xddp.denso.co.jp>
Date: Sun, 20 Apr 2014 22:13:12 +0900
Subject: [PATCH 11/31] ingnore data/wayland.png

Change-Id: I2f74d8e7842c9cb481f7758e0412875c069926f4
---
 Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 869567d..b133fcd 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -650,7 +650,6 @@ BUILT_SOURCES +=					\
 westondatadir = $(datadir)/weston
 dist_westondata_DATA =				\
 	data/wayland.svg			\
-	data/wayland.png			\
 	data/pattern.png			\
 	data/terminal.png			\
 	data/border.png				\
-- 
1.9.3

