From 58da4d5a0b0fd9e8dd512aef0291c66c855b531b Mon Sep 17 00:00:00 2001
From: Nobuhiko Tanibata <NOBUHIKO_TANIBATA@xddp.denso.co.jp>
Date: Tue, 16 Sep 2014 15:21:12 +0900
Subject: [PATCH 30/31] simple-shm: support ivi_surface configure event.

Signed-off-by: Nobuhiko Tanibata <NOBUHIKO_TANIBATA@xddp.denso.co.jp>
---
 clients/simple-shm.c | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/clients/simple-shm.c b/clients/simple-shm.c
index c54b5d7..123d1a2 100644
--- a/clients/simple-shm.c
+++ b/clients/simple-shm.c
@@ -159,6 +159,33 @@ static const struct xdg_surface_listener xdg_surface_listener = {
 	handle_delete,
 };
 
+static void
+handle_ivi_surface_configure(void *data, struct ivi_surface *ivi_surface,
+                             int32_t width, int32_t height)
+{
+	struct window *window = data;
+
+        if (window->buffers[0].buffer) {
+            wl_buffer_destroy(window->buffers[0].buffer);
+            window->buffers[0].buffer = NULL;
+            window->buffers[0].busy = 0;
+        }
+
+        if (window->buffers[1].buffer) {
+            wl_buffer_destroy(window->buffers[1].buffer);
+            window->buffers[1].buffer = NULL;
+            window->buffers[1].busy = 0;
+        }
+
+	window->width = width;
+	window->height = height;
+}
+
+static const struct ivi_surface_listener ivi_surface_listener = {
+        NULL,
+        handle_ivi_surface_configure,
+};
+
 static struct window *
 create_window(struct display *display, int width, int height)
 {
@@ -200,6 +227,10 @@ create_window(struct display *display, int width, int height)
 			fprintf(stderr, "Failed to create ivi_client_surface\n");
 			abort();
 		}
+
+                ivi_surface_add_listener(window->ivi_surface,
+                                         &ivi_surface_listener, window);
+
 	} else {
 		assert(0);
 	}
-- 
1.9.3

