From 58c25a1bee0e65aa273af3958675b9db398851ee Mon Sep 17 00:00:00 2001
From: Igor Socec <igor.socec@pelagicore.com>
Date: Thu, 25 Feb 2016 09:52:15 +0100
Subject: [PATCH] load correct version of libpers_common

Signed-off-by: Igor Socec <igor.socec@pelagicore.com>
---
 config/pclCustomLibConfigFile.cfg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/pclCustomLibConfigFile.cfg.in b/config/pclCustomLibConfigFile.cfg.in
index 185cab4..fe95b18 100644
--- a/config/pclCustomLibConfigFile.cfg.in
+++ b/config/pclCustomLibConfigFile.cfg.in
@@ -1,7 +1,7 @@
 hwinfo @prefix@/lib/libhwinfoperscustom.so init async
 secure @prefix@/lib/libsecureperscustom.so od sync
 custom3 @prefix@/lib/libcustom3perscustom.so init sync
-default @prefix@/lib/libpers_common.so init sync
+default @prefix@/lib/libpers_common.so.0 init sync
 emergency @prefix@/lib/libemergencyperscustom.so od async
 early @prefix@/lib/libearlyperscustom.so od sync
 anInvalidEntry /wrongPath/wrongLib do cnys
-- 
1.9.1

