--- git/systemd/CMakeLists_old.txt	2013-03-12 16:53:37.052664326 +0500
+++ git/systemd/CMakeLists.txt	2013-03-12 16:53:57.052896347 +0500
@@ -46,15 +46,15 @@ if(WITH_SYSTEMD)
     message(STATUS "DLT adaptor udp configuration: APPID=${DLT_ADAPTOR_UDP_APPID} CTID=${DLT_ADAPTOR_UDP_CTID} PORT=${DLT_ADAPTOR_UDP_PORT}" )
     
     
-    if(EXISTS ${SYSTEMD_CONFIGURATIONS_FILES_DIR})
+    #if(EXISTS ${SYSTEMD_CONFIGURATIONS_FILES_DIR})
         install(FILES ${PROJECT_BINARY_DIR}/systemd/dlt.service DESTINATION ${SYSTEMD_CONFIGURATIONS_FILES_DIR} )
         install(FILES ${PROJECT_BINARY_DIR}/systemd/dlt-system.service DESTINATION ${SYSTEMD_CONFIGURATIONS_FILES_DIR} )
         install(FILES ${PROJECT_BINARY_DIR}/systemd/dlt-receive.service DESTINATION ${SYSTEMD_CONFIGURATIONS_FILES_DIR} )
         install(FILES ${PROJECT_BINARY_DIR}/systemd/dlt-example-user.service DESTINATION ${SYSTEMD_CONFIGURATIONS_FILES_DIR} )
         install(FILES ${PROJECT_BINARY_DIR}/systemd/dlt-adaptor-udp.service DESTINATION ${SYSTEMD_CONFIGURATIONS_FILES_DIR} )
         message(STATUS "Unit files will be installed to ${SYSTEMD_CONFIGURATIONS_FILES_DIR} after make install" )
-    else(EXISTS ${SYSTEMD_CONFIGURATIONS_FILES_DIR})
-        message(STATUS "Unit files will not be installed to ${SYSTEMD_CONFIGURATIONS_FILES_DIR} after make install")
-    endif(EXISTS ${SYSTEMD_CONFIGURATIONS_FILES_DIR})
+    #else(EXISTS ${SYSTEMD_CONFIGURATIONS_FILES_DIR})
+    #    message(STATUS "Unit files will not be installed to ${SYSTEMD_CONFIGURATIONS_FILES_DIR} after make install")
+    #endif(EXISTS ${SYSTEMD_CONFIGURATIONS_FILES_DIR})
      
-endif(WITH_SYSTEMD)
\ No newline at end of file
+endif(WITH_SYSTEMD)
